src/fides/api/ctl/sql_models.py

```diff
@@ -313,3 +315,82 @@ class SystemScans(Base):
     for _, sql_model in sql_model_map.items()
     if hasattr(sql_model, "is_default")
 ]
+
+
+class AllowedTypes(str, EnumType):
+    """Allowed types for custom field."""
+
+    string = "string"
+    string_list = "string[]"
+
+
+class ResourceTypes(str, EnumType):
+    """Resource types that can use custom fields."""
+
+    system = "system"
+    data_use = "data use"
+    data_category = "data category"
+    data_subject = "data subject"
+
+
+class CustomFieldValueList(Base):
+    """Allow-list definitions for custom metadata values"""
+
+    __tablename__ = "plus_custom_field_value_list"
+
+    name = Column(String, nullable=False)
+    description = Column(String)
+    allowed_values = Column(ARRAY(String))
+    custom_field_definition = relationship(
+        "CustomFieldDefinition",
+        back_populates="allow_list",
+    )
+
+    UniqueConstraint("name")
+
+
+class CustomFieldDefinition(Base):
+    """Defines custom metadata for resources."""
+
+    __tablename__ = "plus_custom_field_definition"
+
+    name = Column(String, index=True, nullable=False)
+    description = Column(String)
+    field_type = Column(
+        EnumColumn(AllowedTypes),
+        nullable=False,
+    )
+    allow_list_id = Column(String, ForeignKey(CustomFieldValueList.id), nullable=True)
+    resource_type = Column(EnumColumn(ResourceTypes), nullable=False)
+    field_definition = Column(String, index=True)
+    custom_field = relationship(
+        "CustomField",
+        back_populates="custom_field_definition",
+        cascade="delete, delete-orphan",
+    )
+    allow_list = relationship(
+        "CustomFieldValueList",
+        back_populates="custom_field_definition",
+    )
+
+    UniqueConstraint("name", "resource_type")
+
+
+class CustomField(Base):
+    """Custom metadata for resources."""
+
+    __tablename__ = "plus_custom_field"
+
+    resource_type = Column(EnumColumn(ResourceTypes), nullable=False)
+    resource_id = Column(String, index=True, nullable=False)
+    custom_field_definition_id = Column(
+        String, ForeignKey(CustomFieldDefinition.id), nullable=False
+    )
+    value = Column(String)
```

Comment:
just thought about this - how are we supporting a string array for a given custom field instance? i feel like we'll either need another column with type `ARRAY(String)` or we'd need to adjust the `UniqueConstraint` to allow more than one `CustomField` with the same `custom_field_definition_id` on the same `resource_id`.

happy to discuss more offline, this is a bit complex.

Reply:
Discussed in slack. We are going to change `value` to be an `ARRAY(String)`